Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/RewardsReporting.js

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+if (webkit.messageHandlers.rewardsReporting) {
+    install();
+}
+
+function install() {
+    function sendMessage(method, url, data, referrerUrl) {
+        webkit.messageHandlers.rewardsReporting.postMessage({
+            method: method === undefined ? "GET" : method,
+            url: url,
+            data: data === undefined ? null : data,
+            referrerUrl: referrerUrl === undefined ? null : referrerUrl,
+        });
+    }
+    
+    let originalOpen = null;
```

Comment:
We should initialize some of these to what they are being set to later on?

IE: 
```javascript
let originalOpen = XMLHttpRequest.prototype.open;
let originalSend = XMLHttpRequest.prototype.send;
let originalFetch = window.fetch;
let originalSendBeacon = navigator.sendBeacon;
let originalImageSrc = Object.getOwnPropertyDescriptor(Image.prototype, "src");
```

Reply:
Yeah we could do that too, basically just copy pasted from https://github.com/brave/brave-ios/blob/development/Client/Frontend/UserContent/UserScripts/MainFrame/AtDocumentStart/TrackingProtectionStats.js